velox/expression/ConjunctExpr.cpp

```diff
@@ -415,18 +416,89 @@ TypePtr ConjunctCallToSpecialForm::resolveType(
   return ConjunctExpr::resolveType(argTypes);
 }
 
+ExprPtr ConjunctCallToSpecialForm::optimizeAnd(
+    std::vector<ExprPtr>& compiledChildren) {
+  auto numChildren = compiledChildren.size();
+  for (vector_size_t i = numChildren - 1; i >= 0; i--) {
+    if (auto constantExpr = std::dynamic_pointer_cast<exec::ConstantExpr>(
+            compiledChildren[i])) {
+      if (!constantExpr->value()->isNullAt(0)) {
+        if (!constantExpr->value()->as<ConstantVector<bool>>()->valueAt(0)) {
+          return constantExpr;
+        } else {
+          if (i == 0 && compiledChildren.size() == 1) {
+            return constantExpr;
+          } else {
+            compiledChildren.erase(compiledChildren.begin() + i);
+          }
+        }
+      }
+    }
+  }
+
+  if (compiledChildren.size() == 1) {
+    return compiledChildren.front();
+  }
+  return nullptr;
+}
+
+ExprPtr ConjunctCallToSpecialForm::optimizeOr(
+    std::vector<ExprPtr>& compiledChildren) {
+  auto numChildren = compiledChildren.size();
+  for (vector_size_t i = numChildren - 1; i >= 0; i--) {
+    if (auto constantExpr = std::dynamic_pointer_cast<exec::ConstantExpr>(
+            compiledChildren[i])) {
+      if (!constantExpr->value()->isNullAt(0)) {
+        if (constantExpr->value()->as<ConstantVector<bool>>()->valueAt(0)) {
+          return constantExpr;
+        } else {
+          if (i == 0 && compiledChildren.size() == 1) {
+            return constantExpr;
+          } else {
+            compiledChildren.erase(compiledChildren.begin() + i);
+          }
+        }
+      }
+    }
+  }
+
+  if (compiledChildren.size() == 1) {
+    return compiledChildren.front();
+  }
+  return nullptr;
+}
+
+// During expression compilation, the inputs to 'AND', 'OR' conjunct expressions
```

Comment:
This is a nice comment. Please, move it to the header file.

Consider including examples to help readers understand the optimization.

- f(a) AND 1 = 2 => false
- f(b) OR 1 = 1 => true
- f(a) AND 1 = 1 => f(a)
- f(a) OR 1 = 2 => f(a)